### PR TITLE
Documentation additions

### PR DESCRIPTION
**Pull request by npfedwards**
Putting this here as a bit of a work in progress for a bit of feedback before I add to it further.

I've added a couple of examples to the docblocks and changed the callback param to a callback type rather than function.

### CODE REVIEW

**Comment by npfedwards** [comment]
What does `requestHandshake` actually do? The callback seems to have no parameters to give me details on the current installation and if the hive_keychain object isn't there then we'd get an undefined error anyway.

I tested it with
```js
keychain.requestHandshake((response) => {
    console.log(response);
});```

**Comment by stoodkev** [comment]
> What does `requestHandshake` actually do? The callback seems to have no parameters to give me details on the current installation and if the hive_keychain object isn't there then we'd get an undefined error anyway.
> 
> I tested it with
> 
> ```js
> keychain.requestHandshake((response) => {
>     console.log(response);
> });```
> ```

It's just there to make sure that Keychain is there and working properly.

File: documentation/README.md [hunk 1]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
-*   `account` **[String][69]** Hive account to perform the request. If null, user can choose the account from a dropdown (optional, default `null`)
-*   `message` **[String][69]** Message to be signed by the account
-*   `key` **[String][69]** Type of key. Can be 'Posting','Active' or 'Memo'
-*   `callback` **[function][68]** Keychain's response to the request
-*   `rpc` **[String][69]** Override user's RPC settings (optional, default `null`)
-*   `title` **[String][69]** Override "Sign message" title (optional, default `null`)
+*   `account` **[String][95]** Hive account to perform the request. If null, user can choose the account from a dropdown (optional, default `null`)
+*   `message` **[String][95]** Message to be signed by the account
+*   `key` **[String][95]** Type of key. Can be 'Posting','Active' or 'Memo'
+*   `callback` **[requestCallback][96]** Function that handles keychain's response to the request
```
Comment (cedricguillas) [inline documentation/README.md:247]: Can you put a capital K to Keychain to people don't get confused

File: documentation/README.md [hunk 2]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
-*   `authorizedUsername` **[String][69]** Authorized account
-*   `role` **[String][69]** Type of authority. Can be 'Posting','Active' or 'Memo'
-*   `weight` **[number][71]** Weight of the authority
-*   `callback` **[function][68]** Keychain's response to the request
-*   `rpc` **[String][69]** Override user's RPC settings (optional, default `null`)
+*   `account` **[String][95]** Hive account to perform the request
+*   `authorizedUsername` **[String][95]** Authorized account
+*   `role` **[String][95]** Type of authority. Can be 'Posting','Active' or 'Memo'
+*   `weight` **[number][98]** Weight of the authority
+*   `callback` **[requestCallback][96]** Function that handles keychain's response to the request
```
Comment (cedricguillas) [inline documentation/README.md:261]: same here

File: documentation/README.md [hunk 3]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
+*   `rpc` **[String][95]** Override user's RPC settings (optional, default `null`)
+
+##### Examples
+
+```javascript
+// Gives @stoodkev active authority with weight 2
+const keychain = window.hive_keychain
+keychain.requestAddAccountAuthority(username, 'stoodkev', 'Active', 2, (response) => {
```
Comment (cedricguillas) [inline documentation/README.md:269]: Maybe change 'username' to account ? Or something more explicit to show that it is the account that will give the authorization

File: documentation/README.md [hunk 4]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
 
-*   `account` **[String][69]** Hive account to perform the request
-*   `authorizedUsername` **[String][69]** Account to lose authority
-*   `role` **[String][69]** Type of authority. Can be 'Posting','Active' or 'Memo'
-*   `callback` **[function][68]** Keychain's response to the request
-*   `rpc` **[String][69]** Override user's RPC settings (optional, default `null`)
+*   `account` **[String][95]** Hive account to perform the request
+*   `authorizedUsername` **[String][95]** Account to lose authority
+*   `role` **[String][95]** Type of authority. Can be 'Posting','Active' or 'Memo'
+*   `callback` **[requestCallback][96]** Function that handles keychain's response to the request
```
Comment (cedricguillas) [inline documentation/README.md:283]: same here

File: documentation/README.md [hunk 5]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
+*   `rpc` **[String][95]** Override user's RPC settings (optional, default `null`)
+
+##### Examples
+
+```javascript
+// Power up 5 HP
+if (window.hive_keychain) {
+  const keychain = window.hive_keychain;
+  keychain.requestProxy(username, username, '5.000', (response) => {
```
Comment (cedricguillas) [inline documentation/README.md:659]: probably powerDown :)

File: documentation/README.md [hunk 6]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
+*   `callback` **[requestCallback][96]** Function that handles keychain's response to the request
+*   `rpc` **[String][95]** Override user's RPC settings (optional, default `null`)
+
+##### Examples
+
+```javascript
+// Power down 5 HP
+if (window.hive_keychain) {
+  const keychain = window.hive_keychain;
+  keychain.requestProxy(username, '5.000', (response) => {
```
Comment (cedricguillas) [inline documentation/README.md:684]: probably powerDown

File: documentation/README.md [hunk 7]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
-*   `callback` **[function][68]** Keychain's response to the request
-*   `rpc` **[String][69]** Override user's RPC settings (optional, default `null`)
+*   `username` **[String][95]** Hive account to perform the request
+*   `proposal_ids` **[String][95]** Stringified Array of Ids of the proposals to be voted
+*   `approve` **[boolean][102]** Set to true to support the proposal, false to remove a vote
+*   `extensions` **[String][95]** Stringified Array of extensions
+*   `callback` **[requestCallback][96]** Function that handles keychain's response to the request
+*   `rpc` **[String][95]** Override user's RPC settings (optional, default `null`)
+
+##### Examples
```
Comment (cedricguillas) [inline documentation/README.md:775]: maybe here you could add an example of how to remove the vote from a proposal ?

File: documentation/README.md [hunk 8]
```diff
@@ -160,360 +186,695 @@ This function is called to verify Keychain installation on a user's device
...
-*   `recurrence` **[Number][71]** How often will the payment be triggered (in hours).
-*   `executions` **[Number][71]** The times the recurrent payment will be executed.
-*   `callback` **[function][68]** Keychain's response to the request
-*   `rpc` **[String][69]** Override user's RPC settings (optional, default `null`)
+*   `username` **[String][95]** Hive account to perform the request (optional, default `null`)
+*   `to` **[String][95]** Hive account receiving the transfers.
+*   `amount` **[String][95]** amount to be sent on each execution.
+*   `currency` **[String][95]** HIVE or HBD on mainnet.
+*   `memo` **[String][95]** transfer memo
+*   `recurrence` **[Number][98]** How often will the payment be triggered (in hours).
```
Comment (cedricguillas) [inline documentation/README.md:852]: you can add the minimum for those parameters : 
- recurrence has to be minimum 24
- executions has to be minimum 2

**Review by cedricguillas — CHANGES_REQUESTED** [review]
I added some comments, please check them before we can merge. Thank you :)

**Review by cedricguillas — APPROVED** [review]

**Review by stoodkev — APPROVED** [review]

**Comment by stoodkev** [comment]
Thanks for your contribution @npfedwards

**Pull request merged**